sycl/doc/extensions/proposed/sycl_ext_oneapi_fpga_mem.asciidoc

```diff
@@ -0,0 +1,507 @@
+
+`fpga_mem` has reference wrapper-like semantics, and is implicitly convertible to the wrapped type. Of note,
+because {cpp} doesn't allow for overloading of the "dot operator", a `get()`
+member of `fpga_mem` allows a reference to be extracted, to which the usual
+dot operator may be applied.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_FPGA_MEM` to one of the values defined in the table
+below. Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Representation of FPGA memory
+
+[source,c++]
+----
+namespace sycl::ext::intel::experimental {
+template <typename T, typename PropListT = sycl::ext::oneapi::properties<>>
+class fpga_mem {
+  ...
+----
+
+`fpga_mem` is a class template, parameterized by the type of the underlying
+allocation _T_, and a list of properties _PropListT_. The type of the
+allocation _T_ also encodes the size of the allocation.
+
+`fpga_mem` initializes the underlying object `T` with the values passed into its constructor.
+
+Properties may be specified for a `fpga_mem` to provide semantic
+modification or optimization hint information to the compiler. See the section
+below for a list of the properties that are allowed.
+
+[NOTE]
+====
+
+`fpga_mem` has similar semantics to a reference wrapper. An fpga_mem templated on a type T can implicitly convert to T, thereby allowing it to be used in most places that an object of type T could be used. The dot operator
+(`operator.`) cannot be overloaded, so a `get()` member is provided to allow a
+reference to be extracted directly when needed.
+====
+
+=== FPGA Memory Interface
+
+The section below and the table following describe the constructors and member functions for `fpga_mem`.
+
+[source,c++]
+----
+namespace sycl::ext::intel::experimental {
+
+  template <typename T, typename PropertyListT = properties<>>
+  class fpga_mem {
+  public:
+    template <typename... S> constexpr fpga_mem(S... args);
+
+    template <access::decorated IsDecorated>
+    multi_ptr<T, access::address_space::private_space, IsDecorated>
+    get_multi_ptr() noexcept;
+
+    template <access::decorated IsDecorated>
+    multi_ptr<const T, access::address_space::private_space, IsDecorated>
+    get_multi_ptr() const noexcept;
```

Comment:
Is this such a common case that we need to make a convenience member function for it?  It seems to me that in the rare cases where a user needs this they can construct the multi_ptr manually.